qutebrowser/mainwindow/statusbar/settings.py

```diff
@@ -0,0 +1,130 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2018 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Generic settings status displayed in the statusbar."""
+
+from qutebrowser.config import config, configtypes
+from qutebrowser.mainwindow.statusbar import textbase
+from qutebrowser.utils import objreg, message
+
+
+class BooleanSettings(textbase.TextBase):
+
+    """Boolean settings state displayed in the statusbar."""
+
+    # A list of indicator, setting_name pairs.
+    _config = []
```

Comment:
This should be set in `__init__` instead, or all instances of the class will share it, which might be quite confusing.